### PR TITLE
ncclient j2config support

### PR DESCRIPTION
**Pull request by rhwendt**
This PR addresses some issues I was facing when attempting to push config using ncclient. Below is a summary of the issues I had when using this library.

- editconfig would return `Unserializable return value` from redis jobs
- lack of jinja2 templating when using ncclient

Here is a summary of the changes I am proposing.

- Move `render_json` boolean out of the `args` body making it an optional component. This allows me to render json from the reponses of getconfig and setconfig when using the ncclient library.
- `render_json` works for both getconfig and setconfig respones in the ncclient library.
- Made `args` in `NcclientSetConfig` model optional.
- Made `j2config` in `NcclientSetConfig` model optional.
- If you rendered j2config for ncclient it will add it to the `kwargs['args']['config']` dict which gets passed to ncclient.
- `ncclient_drvr.ncclien:editconfig` formats the response as xml
- added a ncclient template for pytest
- added a new method for setconfig mark for pytest `test_setconfig_ncclient_j2()`
- fixed references to `render_json` in test files.

Side note: Do you have issues with github actions failing randomly for no reason? I built it several times locally and they passed but when I ran the github actions they would fail on different errors each time and eventually it passed after running more than once?

### CODE REVIEW

**Comment by tbotnz** [comment]
thanks for the pr @rhwendt will review tonight

**Comment by tbotnz** [comment]
Would it be possible to keep _render_json_ within _args_ so that we are not changing the public API for **all** users at this point.
Whilst i completely understand the reasoning for moving it out of args i am just mindful of the impact on other users at this point. 

This is my only concern with the PR

**Comment by rhwendt** [comment]
Yeah that's a fair concern. I can rework this so I don't affect everyone.

**Comment by rhwendt** [comment]
Still working on this just been tied up with other things.

**Comment by rhwendt** [comment]
Made `config` and `target` optional arguments in `NcclientSendConfigArgs`

**config**
This allows me to render j2config and add it to the args body after calling `exec_config`. I could have pre-rendered it using one of the other URI's but this seems like a hassle when the functionality is already in `exec_config`. If `config` is omitted ncclient will still throw an exception vs pydantic throwing one.

**target**
Already defaults to `candidate` in ncclient.

Let me know if you have any concerns.

P.S. sorry about the typo's :smile:

**Comment by rhwendt** [comment]
@tbotnz This is complete, but let me know if you have any concerns. Thanks!

File: netpalm/backend/core/models/models.py [hunk 1]
```diff
@@ -168,6 +159,7 @@ class Config:
                     "use_textfsm": True
                 },
                 "queue_strategy": "pinned",
+                "render_json": True,
```
Comment (tbotnz) [inline netpalm/backend/core/models/models.py:162]: plz move under args

**Review by tbotnz — CHANGES_REQUESTED** [review]
Thanks for fixing this up @rhwendt much appreciated, sorry for the delay in getting back to you on this.

Can only see one minor issue below, if you fix that then happy to merge

**Pull request merged**